chickadee/processes/wps_ca.py

```diff
@@ -0,0 +1,177 @@
+            "write_files": 80,
+            "build_output": 95,
+            "complete": 100,
+        }
+
+        inputs = [
+            gcm_file,
+            obs_file,
+            varname,
+            num_cores,
+            end_date,
+            LiteralInput(
+                "indices",
+                "Indices File",
+                abstract="File name to store indices of analogue times steps (suffix .txt)",
+                data_type="string",
+            ),
+            LiteralInput(
+                "weights",
+                "Weights File",
+                abstract="File name to store weights of analogues (suffix .txt)",
+                data_type="string",
+            ),
+            log_level,
+        ]
+
+        outputs = [
+            ComplexOutput(
+                "indices_file",
+                "Indices File",
+                abstract="File path with analogue indices",
+                supported_formats=[FORMATS.TEXT],
+            ),
+            ComplexOutput(
+                "weights_file",
+                "Weights File",
+                abstract="File path with analogue weights",
+                supported_formats=[FORMATS.TEXT],
+            ),
+        ]
+
+        super(CA, self).__init__(
+            self._handler,
+            identifier="ca",
+            title="CA",
+            abstract="Constructed Analogue (CA) downscaling algorithm",
+            keywords=["constructed", "analogue", "downscaling"],
+            metadata=[
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            version="0.1.0",
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def get_input_type(self, input_name, request):
+        if "file" in input_name:
+            return request.inputs[input_name][0].file
+        else:
+            return str(request.inputs[input_name][0].data)
+
+    def write_list_to_file(self, list_, filename):
+        with open(filename, "w") as file_:
+            for line in list_:
+                for item in line:
+                    file_.write(f"{item}\n")
+
+    def _handler(self, request, response):
+        loglevel = request.inputs["loglevel"][0].data
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+
+        (
+            gcm_file,
+            obs_file,
+            varname,
+            num_cores,
+            end_date,
+            indices,
+            weights,
+            log_level,
+        ) = [self.get_input_type(input_name, request) for input_name in request.inputs]
+        [os.path.join(self.workdir, file_) for file_ in [indices, weights]]
```

Comment:
It seems like you've created the equivalent of the function [collect_args](https://github.com/pacificclimate/chickadee/blob/i12-qdm/chickadee/utils.py#L33) that I am modifying to return the values dynamically. I haven't tried this in Docker container yet but expecting it to work since a similar method was used in `osprey`.